addons/mail/static/src/core/messaging_service.js

```diff
@@ -205,6 +205,34 @@ export class Messaging {
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this.threadService.insert({
+                            id: notif.payload.id,
+                            model: "discuss.channel",
+                        });
+                        this.store.discuss.starred.messages =
```

Comment:
For this case in particular, if there was a starred message found, the counter should be decremented by one here.